libraries/persistent_store/src/format.rs

```diff
@@ -0,0 +1,907 @@
+    ///
+    /// Notation: `E`. We have `E <= MAX_CYCLE`.
+    pub fn max_page_erases(&self) -> usize {
+        self.max_page_erases
+    }
+
+    /// The maximum key.
+    pub fn max_key(&self) -> usize {
+        MAX_KEY
+    }
+
+    /// The maximum number of updates per transaction.
+    pub fn max_updates(&self) -> usize {
+        MAX_UPDATES
+    }
+
+    /// The size of a virtual page in words.
+    ///
+    /// A virtual page is stored in a physical page after the page header.
+    ///
+    /// Notation: `Q`. We have `6 <= Q <= MAX_VIRT_PAGE_SIZE`.
+    pub fn virt_page_size(&self) -> usize {
+        self.page_size() / self.word_size() - CONTENT_WORD
+    }
+
+    /// The maximum length in bytes of a user payload.
+    ///
+    /// We have `20 <= self.max_value_len() <= MAX_VALUE_LEN`.
+    pub fn max_value_len(&self) -> usize {
+        min(
+            (self.virt_page_size() - 1) * self.word_size(),
+            MAX_VALUE_LEN,
+        )
+    }
+
+    /// The maximum prefix length in words.
+    ///
+    /// A prefix is the first words of a virtual page that belong to the last entry of the previous
+    /// virtual page. This happens because entries may overlap up to 2 virtual pages.
+    ///
+    /// Notation: `M`. We have `5 <= M < Q`.
+    pub fn max_prefix_len(&self) -> usize {
+        self.bytes_to_words(self.max_value_len())
+    }
+
+    /// The total virtual capacity in words.
+    ///
+    /// Notation: `V`. We have `V = (N - 1) * (Q - 1) - M`.
+    pub fn virt_size(&self) -> usize {
+        (self.num_pages() - 1) * (self.virt_page_size() - 1) - self.max_prefix_len()
+    }
+
+    /// The total user capacity in words.
+    ///
+    /// Notation: `C`. We have `C = V - N = (N - 1) * (Q - 2) - M - 1`.
+    pub fn total_capacity(&self) -> usize {
+        // From the virtual capacity, we reserve N - 1 words for `Erase` entries and 1 word for a
+        // `Clear` entry.
+        self.virt_size() - self.num_pages()
+    }
+
+    /// The total virtual lifetime in words.
+    ///
+    /// Notation: `L`. We have `L = (E * N + N - 1) * Q`.
+    pub fn total_lifetime(&self) -> Position {
+        Position::new(self, self.max_page_erases(), self.num_pages() - 1, 0)
+    }
+
+    /// Returns the word position of the first entry of a page.
+    ///
+    /// The init info of the page must be provided to know where the first entry of the page
+    /// starts.
+    pub fn page_head(&self, init: InitInfo, page: usize) -> Position {
+        Position::new(self, init.cycle, page, init.prefix)
+    }
+
+    /// Returns the storage index of the init info of a page.
+    pub fn index_init(&self, page: usize) -> StorageIndex {
+        let byte = INIT_WORD * self.word_size();
+        StorageIndex { page, byte }
+    }
+
+    /// Parses the init info of a page from its storage representation.
+    pub fn parse_init(&self, word: &[u8]) -> StoreResult<WordState<InitInfo>> {
+        let word = slice_to_word(word);
```

Comment:
I don't see the callers of this API yet, but is it better to have this API take an unsized slice `&[u8]` as input and do the `slice_to_word` check (which can panic) internally, or to have this API take a `&WORD` type and have the caller be responsible for converting slices to words? I guess it'd be easier to prove the absence of panics if it's the responsibility of the caller.